modules/rabbitmq/src/main/java/org/testcontainers/containers/RabbitMQContainer.java

```diff
@@ -0,0 +1,358 @@
+     */
+    public String getAmqpsUrl() {
+        return String.format("amqps://" + getContainerIpAddress() + ":" + getMappedPort(DEFAULT_AMQPS_PORT));
+    }
+
+    /**
+     * @return URL of the HTTP management endpoint.
+     */
+    public String getHttpUrl() {
+        return String.format("http://" + getContainerIpAddress() + ":" + getMappedPort(DEFAULT_HTTP_PORT));
+    }
+
+    /**
+     * @return URL of the HTTPS management endpoint.
+     */
+    public String getHttpsUrl() {
+        return String.format("https://" + getContainerIpAddress() + ":" + getMappedPort(DEFAULT_HTTPS_PORT));
+    }
+
+    /**
+     * Sets the password for the admin (default is <pre>guest</pre>)
+     *
+     * @param adminPassword The admin password.
+     * @return This container.
+     */
+    public S withAdminPassword(final String adminPassword) {
+        this.adminPassword = adminPassword;
+        return self();
+    }
+
+    public enum SslVerification {
+        VERIFY_NONE("verify_none"), VERIFY_PEER("");
+
+        SslVerification(String value) {
+            this.value = value;
+        }
+
+        private final String value;
+    }
+
+    public S withSSL(
+            final MountableFile keyFile,
+            final MountableFile certFile,
+            final MountableFile caFile,
+            SslVerification verify,
+            boolean failIfNoCert,
+            int verificationDepth) {
+
+        return withSSL(keyFile, certFile, caFile, verify, failIfNoCert)
+                .withEnv("RABBITMQ_SSL_DEPTH", String.valueOf(verificationDepth));
+    }
+
+    public S withSSL(
+            final MountableFile keyFile,
+            final MountableFile certFile,
+            final MountableFile caFile,
+            SslVerification verify,
+            boolean failIfNoCert) {
+
+        return withSSL(keyFile, certFile, caFile, verify)
+                .withEnv("RABBITMQ_SSL_FAIL_IF_NO_PEER_CERT", String.valueOf(failIfNoCert));
+    }
+
+    public S withSSL(
+            final MountableFile keyFile,
+            final MountableFile certFile,
+            final MountableFile caFile,
+            SslVerification verify) {
+
+        return withEnv("RABBITMQ_SSL_CACERTFILE", "/etc/rabbitmq/ca_cert.pem")
+                .withEnv("RABBITMQ_SSL_CERTFILE", "/etc/rabbitmq/rabbitmq_cert.pem")
+                .withEnv("RABBITMQ_SSL_KEYFILE", "/etc/rabbitmq/rabbitmq_key.pem")
+                .withEnv("RABBITMQ_SSL_VERIFY", verify.value)
+                .withCopyFileToContainer(certFile, "/etc/rabbitmq/rabbitmq_cert.pem")
+                .withCopyFileToContainer(caFile, "/etc/rabbitmq/ca_cert.pem")
+                .withCopyFileToContainer(keyFile, "/etc/rabbitmq/rabbitmq_key.pem");
+    }
+
+    public S withPluginsEnabled(String... pluginNames) {
+        List<String> command = new ArrayList<>(asList("rabbitmq-plugins", "enable"));
+        command.addAll(asList(pluginNames));
+        command.add("--offline");
```

Comment:
this does not seem to change anything, it only creates a collection without changing the state of this container

Reply:
Hmm, that's very true. Must be something wrong with my test, as it's passing. I'll review it.

Reply:
Fixed the test and found the bug. :+1: